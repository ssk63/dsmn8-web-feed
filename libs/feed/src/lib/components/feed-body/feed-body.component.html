<div class="feed-body">
  <div class="feed-body__header">
    <h3 class="feed-body__header-title" *ngIf="name">
      {{ name }}
    </h3>
  </div>
  <div class="feed-body__content" *ngIf="content" [innerText]="content"></div>
  <div class="feed-body__tags">
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
      >#marketing</a
    >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
      >#marketingstrategy</a
    >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
      >#employeeadvocacy</a
    >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
      >#employeeinfluencers</a
    >
  </div>
  <div class="feed-body__image" *ngIf="imgUrl">
    <img
      [attr.src]="imgUrl"
      alt="Europe's most active mechanical or industrial engineering professionals on social"
    />
  </div>
</div>
