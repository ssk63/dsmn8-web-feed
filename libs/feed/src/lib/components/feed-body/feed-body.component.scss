.feed-body {
  border: 1px solid #eceef1;
  border-radius: 10px;
  &__header,
  &__content,
  &__tags {
    padding: 0 20px;
  }
  &__tags {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin: 20px 0;
    &-link {
      padding-right: 15px;
    }
  }
  &__image {
    display: flex;
    img {
      width: 100%;
      height: 100%;
    }
  }
}
