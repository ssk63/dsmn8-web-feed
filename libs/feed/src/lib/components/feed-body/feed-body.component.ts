import { Component, Input, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'dsmn8-feed-body',
  templateUrl: './feed-body.component.html',
  styleUrls: ['./feed-body.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class FeedBodyComponent {
  @Input() name: string | undefined;
  @Input() content: string | undefined;
  @Input() imgUrl: string | undefined;
}
