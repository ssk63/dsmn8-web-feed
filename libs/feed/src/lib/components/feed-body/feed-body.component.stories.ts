import { moduleMetadata, Story, Meta } from '@storybook/angular';
import { mockFeed } from '../../models';
import { FeedBodyComponent } from './feed-body.component';

export default {
  title: 'Pages/Feed/FeedBodyComponent',
  component: FeedBodyComponent,
  decorators: [
    moduleMetadata({
      imports: [],
    }),
  ],
  argTypes: {
    name: {
      name: 'name',
      description: 'Sets the feed post title.',
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
    content: {
      name: 'content',
      description: 'Sets the feed post text.',
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
    imgUrl: {
      name: 'imgUrl',
      description: 'Sets the feed post image.',
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
    videoUrl: {
      name: 'videoUrl',
      description: 'Sets the feed post video.',
      control: {
        type: 'text',
      },
      table: {
        category: 'Inputs',
      },
    },
  },
  parameters: {
    jest: 'feed-body.component',
  }
} as Meta<FeedBodyComponent>;

const Template: Story<FeedBodyComponent> = (args: FeedBodyComponent) => ({
  props: args,
});

export const FeedTitle = Template.bind({});
FeedTitle.args = {
  name: mockFeed[0].name,
};

export const FeedContent = Template.bind({});
FeedContent.args = {
  content: mockFeed[0].content,
};

export const FeedImage = Template.bind({});
FeedImage.args = {
  imgUrl: mockFeed[0].imgUrl,
};

export const FeedVideo = Template.bind({});
FeedVideo.args = {
  videoUrl: mockFeed[3].videoUrl,
};
