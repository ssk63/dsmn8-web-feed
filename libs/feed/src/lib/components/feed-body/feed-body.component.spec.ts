import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FeedBodyComponent } from './feed-body.component';

describe('FeedBodyComponent', () => {
  let component: FeedBodyComponent;
  let fixture: ComponentFixture<FeedBodyComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FeedBodyComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(FeedBodyComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
