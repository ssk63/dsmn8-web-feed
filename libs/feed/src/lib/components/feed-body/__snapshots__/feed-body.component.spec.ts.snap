// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedBodyComponent should render the feed content when the content is exists 1`] = `
<div
  class="feed-body"
>
  <div
    class="feed-body__header"
  >
    
  </div>
  <div
    class="feed-body__content"
  />
  
  <div
    class="feed-body__tags"
  >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketing
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketingstrategy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeadvocacy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeinfluencers
    </a>
  </div>
  
</div>
`;

exports[`FeedBodyComponent should render the feed image when the imgUrl is exists 1`] = `
<div
  class="feed-body"
>
  <div
    class="feed-body__header"
  >
    
  </div>
  
  <div
    class="feed-body__tags"
  >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketing
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketingstrategy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeadvocacy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeinfluencers
    </a>
  </div>
  <div
    class="feed-body__image"
  >
    <img
      alt="Europe's most active mechanical or industrial engineering professionals on social"
      src="https://dsmn8.imgix.net/dsmn8/image/PcdTvyy0wRnhpaC5sCEjXbgca.jpeg?w=1200"
    />
  </div>
  
</div>
`;

exports[`FeedBodyComponent should render the feed title when the name is exists 1`] = `
<div
  class="feed-body"
>
  <div
    class="feed-body__header"
  >
    <h3
      class="feed-body__header-title"
    >
       DSMN8 - The Employee Influence Platform 
    </h3>
    
  </div>
  
  <div
    class="feed-body__tags"
  >
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketing
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #marketingstrategy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeadvocacy
    </a>
    <a
      class="feed-body__tags-link"
      href="https://www.linkedin.com/"
      target="_blank"
    >
      #employeeinfluencers
    </a>
  </div>
  
</div>
`;
